sql-plugin/src/test/scala/com/nvidia/spark/rapids/HostAllocSuite.scala

```diff
@@ -704,18 +704,19 @@ class HostAllocSuite extends AnyFunSuite with BeforeAndAfterEach with
           val w = SpillableHostBuffer.apply(HostAlloc.alloc(1024, preferPinned),
             1024, SpillPriorities.ACTIVE_BATCHING_PRIORITY)
 
-          // Why can't we just use Thread.sleep here? Because TIMED_WAITING is a special state
-          // treated as BUFN in spark_resource_adaptor::is_thread_bufn_or_above, this will cause
-          // issues like https://github.com/NVIDIA/spark-rapids/issues/12883
-          val startTime = System.currentTimeMillis
-          while (System.currentTimeMillis < startTime + 100) {
-            Thread.`yield`() // when yielding, the thread is in RUNNABLE state
-          }
-
           // let's say we will do sth with the shuffle data read, e.g. concat them
           // for simplicity, we just have one SpillableHostBuffer as input, in real case there
           // should be multiple SpillableHostBuffer as input
           withResource(w.getHostBuffer()) { _ =>
+
+            // Why can't we just use Thread.sleep here? Because TIMED_WAITING is a special state
```

Comment:
This makes the race much less likely to happen, so I am willing to merge this as is, but technically the race is still possible. It should be very unlikely that two threads launched at the same time will diverge by more than 100 ms, but in a CPU constrained environment it might be possible. It also is not great that we have to have a 100 ms sleep. It would be better to have coordination, perhaps with a count down latch to verify that both threads got to the same point at the same time. But also like the comment says if we have one thread block waiting on the other we might end up with some other issues.